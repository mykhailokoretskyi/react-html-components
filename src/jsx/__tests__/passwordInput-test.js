jest.mock('react/lib/ReactDefaultInjection');
import React from 'react';
import PasswordInput from '../passwordInput.jsx';
import renderer from 'react/lib/ReactTestRenderer';

const PasswordInputWithListeners = (
    <PasswordInput required={true}
                    label="test label"
                    name="test_input"
                    validate={true}
                    errorMessage="wrong"
                    successMessage="correct"
                    id="test"
                    changeCallback={function(){}}
                    mouseEnterCallback={function(){}}
                    mouseLeaveCallback={function(){}}/>
);

const PasswordInputWithoutListeners = (
    <PasswordInput required={true}
                label="test label"
                name="test_input"
                id="test"/>
);

const PasswordInputWithValue = (
    <PasswordInput label="test label"
                name="test_input"
                id="test"
                value="value"
                type="text"/>
);

describe('PasswordInput', () => {
    it('with listeners matches snapshot', () => {
        const component = renderer.create(PasswordInputWithListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('without listeners matches snapshot', () => {
        const component = renderer.create(PasswordInputWithoutListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('with value matches snapshot', () => {
        const component = renderer.create(PasswordInputWithValue);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
