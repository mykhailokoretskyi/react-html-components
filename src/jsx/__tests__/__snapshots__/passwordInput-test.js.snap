exports[`PasswordInput with listeners matches snapshot 1`] = `
<div
  className="input-field col s12">
  <input
    checked={false}
    className="validate "
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onClick={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={true}
    type="password"
    value="" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error="wrong"
    data-success="correct"
    htmlFor="test" />
</div>
`;

exports[`PasswordInput with value matches snapshot 1`] = `
<div
  className="input-field col s12">
  <input
    checked={false}
    className=""
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onClick={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={false}
    type="password"
    value="value" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error=""
    data-success=""
    htmlFor="test" />
</div>
`;

exports[`PasswordInput without listeners matches snapshot 1`] = `
<div
  className="input-field col s12">
  <input
    checked={false}
    className=""
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onClick={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={true}
    type="password"
    value="" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error=""
    data-success=""
    htmlFor="test" />
</div>
`;
